eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/zmq/ZMQActor.scala

```diff
@@ -47,26 +48,33 @@ class ZMQActor(address: String, connected: Option[Promise[Boolean]] = None) exte
   implicit val ec: ExecutionContext = context.system.dispatcher
 
   // we check messages in a non-blocking manner with an interval, making sure to retrieve all messages before waiting again
-  def checkEvent: Unit = Option(Event.recv(monitor, ZMQ.DONTWAIT)) match {
+  @tailrec
+  final def checkEvent: Unit = Option(Event.recv(monitor, ZMQ.DONTWAIT)) match {
     case Some(event) =>
       self ! event
       checkEvent
-    case None =>
-      context.system.scheduler.scheduleOnce(1 second)(checkEvent)
+    case None => ()
   }
 
-  def checkMsg: Unit = Option(ZMsg.recvMsg(subscriber, ZMQ.DONTWAIT)) match {
+  @tailrec
+  final def checkMsg: Unit = Option(ZMsg.recvMsg(subscriber, ZMQ.DONTWAIT)) match {
     case Some(msg) =>
       self ! msg
       checkMsg
-    case None =>
-      context.system.scheduler.scheduleOnce(1 second)(checkMsg)
+    case None => ()
   }
 
-  checkEvent
-  checkMsg
+  self ! 'checkEvent
+  self ! 'checkMsg
 
   override def receive: Receive = {
+    case 'checkEvent =>
+      checkEvent
+      context.system.scheduler.scheduleOnce(1 second, self ,'checkEvent)
```

Comment:
@sstone I'm just curious, what are the benefits of scheduling a queue check this way?